### DISCUSSION TITLE
Enable inference of types based on the dot ('.') operator

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1313] [Type System]
Description :
Extending inference to dot operator as in ```myVar.a```

Usage :
Allows function to use parameter with the dot operator without first defining the type of the parameter, anonymous record benefits mostly from the feature (today we can get around the system by wrapping the anonymous record in a class type and pass that type to the function, since we previously wrapped the anonymous record in the type, we can use the anonymous record in the type that's passed into a function at a later point. The down side is the wrapper can't be a struct type otherwise you had to specify in the constructor of the anonymous record's type

Example :
```F#
let f (obj) =
    obj.a = 0
```

**Comment by Xyncgas** [comment]
You can imagine writing a function to handle an anonymous record becomes annoying when there is so many stuffs inside the AR so you are required to write everything in every function's parameter

Imagine how it becomes harder to write functions for processing anonymous records if the record has nested AR

**Comment by jwosty** [comment]
There's gotta be an existing suggestion for this, but I can't find it

**Reply by Xyncgas** [reply]
https://github.com/fsharp/fslang-suggestions/discussions/1156#discussioncomment-2966045
